### PR TITLE
Remove linux-isms in the set-version script.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
https://github.com/brendanhay/amazonka/issues/421

### CODE REVIEW

**Comment by tmcgilchrist** [comment]
This change is fixing the linux-isms so the script can be run on OSX with the BSD version of tools.

File: script/set-version [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-semantic="$2"
-
-if [ -z "$pvp" ] || [ -z "$semantic" ]; then
+if [ $# -ne 2 ]; then
     echo "Usage: bump-version <VERSION> <DEPENDENCY>"
     exit 1
 fi
 
+pvp="$1"
+semantic="$2"
```
Comment (tmcgilchrist) [inline script/set-version]: The check around arguments wasn't working in my shell, I think checking the length of arguments and printing usage before failing is the more robust solution.

File: script/set-version [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+
 specs=(
  amazonka/amazonka.cabal
  core/amazonka-core.cabal
  examples/amazonka-examples.cabal
  gen/amazonka-gen.cabal
  test/amazonka-test.cabal
- amazonka-s3-encryption/amazonka-s3-encryption.cabal
```
Comment (tmcgilchrist) [inline script/set-version]: I had a query about this set of cabal files and why these particular ones are included.
My intuition is that it should include all cabal files. @brendanhay

File: script/set-version [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 echo "Bumping version to $pvp and dependencies to == $semantic ..."
 
 for cabal in ${specs[@]}; do
-    sed -i"" "s/^\(version: *\).*$/\1$pvp/" $cabal
-    sed -i"" "s/^\(.*amazonka-[a-z0-9]* *== *\)\([0-9\.]\+\*\)$/\1$semantic.*/" $cabal
-    sed -i"" "s/^\(.*amazonka *== *\)\([0-9\.]\+\*\)$/\1$semantic.*/" $cabal
+    sed -e "s/^\(version: *\).*$/\1$pvp/;"  \
+        -e "s/^\(.*amazonka-[a-z0-9]* *== *\)\([0-9\.]\+\*\)$/\1$semantic.*/" \
+        -e "s/^\(.*amazonka *== *\)\([0-9\.]\+\*\)$/\1$semantic.*/" $cabal > $cabal.new
+    mv $cabal.new $cabal
```
Comment (tmcgilchrist) [inline script/set-version]: Echoing out to a new file then using mv gets around the incompatibilities with `sed` and inlace edits.

File: script/set-version (hunk 2 above)
Comment (brendanhay) [reply script/set-version]: The `set-version` script deals with the hand-rolled `cabal` files, not the generated ones. The flow is to run `set-version` to bump the listed `cabal` file versions, and then run the generator which will discover those versions via:

- [cabal.ede](https://github.com/brendanhay/amazonka/blob/afeb929c758c772fe83692a54978cf7bb05a9de5/gen/template/cabal.ede#L58) - Template for generated `amazonka-*/amazonka-*.cabal` files.
- [Makefile](https://github.com/brendanhay/amazonka/blob/afeb929c758c772fe83692a54978cf7bb05a9de5/gen/Makefile#L9-L15) - which passes the versions from `amazonka` and `amazonka-core` into the generator.

Part of the reason for this (and why `set-version` takes two version numbers) is to allow dependency versions to differ/go out of sync. So far I've only used this a couple of times when pushing out a hotfix to either `amazonka-core` or `amazonka-ec2`, separately.

File: script/set-version (hunk 3 above)
Comment (brendanhay) [reply script/set-version]: :+1:

File: script/set-version (hunk 2 above)
Comment (brendanhay) [inline script/set-version]: Any reason to remove `amazonka-s3-encryption`? It similarly needs to be version bumped and is somewhat confusing, not auto-generated.

Comment (tmcgilchrist) [reply script/set-version]: No reason, I'll add it back in.

Comment (tmcgilchrist) [reply script/set-version]: 👍 thanks for the explanation.

**Pull request merged**